pkg/providers/validator/validate.go

```diff
@@ -10,9 +10,24 @@ import (
 	"github.com/aws/eks-anywhere/pkg/providers"
 )
 
-// TODO: update vsphere, cloudstack, tinkerbell validators to use this common validation method.
-func ValidateControlPlaneIpUniqueness(cluster *v1alpha1.Cluster, netClient networkutils.NetClient) error {
-	ip := cluster.Spec.ControlPlaneConfiguration.Endpoint.Host
+// DefaultIPValidator defines the struct for the default IPValidator.
+type DefaultIPValidator struct{}
+
+// IPValidator defines an interface for the methods to validate the control plane IP.
+type IPValidator interface {
+	ValidateControlPlaneIPUniqueness(cluster v1alpha1.ClusterSpec) error
+}
+
+// NewIPValidator initializes a new IPValidator object.
+func NewIPValidator() IPValidator {
+	return DefaultIPValidator{}
+}
+
+// ValidateControlPlaneIPUniqueness checks whether or not the control plane endpoint defined
+// in the cluster spec is available.
+func (v DefaultIPValidator) ValidateControlPlaneIPUniqueness(spec v1alpha1.ClusterSpec) error {
+	netClient := &networkutils.DefaultNetClient{}
```

Comment:
This makes this method untestable. I think the network client should be a dependency injected on construction or at least a dependency that can updated from the tests, even if you default to the DefaultNetClient on construction